utils/create_dependency_mapping.py

```diff
@@ -20,12 +32,12 @@ def topological_sort(dependencies: dict):
     while len(graph) > 0:
         # Find the nodes with 0 out-degree
         leaf_nodes = {node for node in graph if len(graph[node]) == 0}
-        # Add them to the list
-        sorting_list += list(leaf_nodes)
+        # Add them to the list as next level
+        sorting_list.append([name_mapping[node] for node in leaf_nodes])
```

Comment:
Instead of collecting a flattened list, we preserve a hierarchical order, allowing us to process models at the same "level" of hierarchy in parallel